aeron-samples/src/main/java/io/aeron/cluster/ConsensusModuleSnapshotPendingServiceMessagesPatch.java

```diff
@@ -0,0 +1,570 @@
+
+        public void onLoadPendingMessage(
+            final long clusterSessionId,
+            final DirectBuffer buffer,
+            final int offset,
+            final int length)
+        {
+            tempBuffer.putBytes(0, buffer, offset, length);
+            sessionMessageHeaderEncoder
+                .wrap(tempBuffer, MessageHeaderEncoder.ENCODED_LENGTH)
+                .clusterSessionId(nextClusterSessionId++);
+
+            writeToASnapshot(tempBuffer, 0, length);
+        }
+
+        public void onLoadClusterSession(
+            final long clusterSessionId,
+            final long correlationId,
+            final long openedLogPosition,
+            final long timeOfLastActivity,
+            final CloseReason closeReason,
+            final int responseStreamId,
+            final String responseChannel,
+            final DirectBuffer buffer,
+            final int offset,
+            final int length)
+        {
+            writeToASnapshot(buffer, offset, length);
+        }
+
+        public void onLoadTimer(
+            final long correlationId,
+            final long deadline,
+            final DirectBuffer buffer,
+            final int offset,
+            final int length)
+        {
+            writeToASnapshot(buffer, offset, length);
+        }
+
+        public void onLoadPendingMessageTracker(
+            final long nextServiceSessionId,
+            final long logServiceSessionId,
+            final int pendingMessageCapacity,
+            final int serviceId,
+            final DirectBuffer buffer,
+            final int offset,
+            final int length)
+        {
+            writeToASnapshot(buffer, offset, length);
+        }
+
+        public void onLoadEndSnapshot(final DirectBuffer buffer, final int offset, final int length)
+        {
+            writeToASnapshot(buffer, offset, length);
+        }
+
+        private void writeToASnapshot(
+            final DirectBuffer buffer,
+            final int offset,
+            final int length)
+        {
+            long result;
+            while ((result = snapshotPublication.offer(buffer, offset, length)) < 0)
+            {
+                if (result == CLOSED ||
+                    result == NOT_CONNECTED ||
+                    result == MAX_POSITION_EXCEEDED)
+                {
+                    throw new ClusterException("can't offer into a snapshot: " + result);
+                }
+                Thread.yield();
+            }
+        }
+    }
+
+    private static final class RecordingSignalInfo implements RecordingSignalConsumer
+    {
+        long recordingId;
+        RecordingSignal signal;
+
+        public void onSignal(
+            final long controlSessionId,
+            final long correlationId,
+            final long recordingId,
+            final long subscriptionId,
+            final long position,
+            final RecordingSignal signal)
+        {
+            this.recordingId = recordingId;
```

Comment:
Be careful of overwriting existing capture `recordingId` as some signals have `-1` when not is relevant.

Reply:
Fixed.